src/Type/Definition/Type.php

```diff
@@ -144,11 +145,18 @@ public static function getStandardTypes(): array
 
     /**
      * @param array<ScalarType> $types
+     * @param bool $partial Whether to replace only types listed in $types or all standard types
      *
      * @throws InvariantViolation
      */
-    public static function overrideStandardTypes(array $types): void
+    public static function overrideStandardTypes(array $types, bool $partial = true): void
     {
+        self::reset();
+
+        if (! $partial) {
+            static::$standardTypes = null;
+        }
```

Comment:
I don't see why it would be necessary to force reinstantiation of standard types that are not being overridden.